opennms-config-model/src/main/java/org/opennms/netmgt/config/eventd/EventdConfiguration.java

```diff
@@ -162,6 +192,14 @@ public Integer getUDPPort() {
     public void setUDPPort(final Integer UDPPort) {
         m_udpPort = ConfigUtils.assertNotNull(UDPPort, "UDPPort");
     }
+    
+    public Optional<Integer> getNumThreads() {
+    	return Optional.ofNullable(m_threads);
+    }
+
+    public void setNumThreads(final Integer numThreads) {
+        m_threads = ConfigUtils.assertNotNull(numThreads, "threads");
```

Comment:
Why do we not allow this field to be null?

Reply:
We have maintained same legacy of opennms i.e syslogd it self which is actually kept as not null  for example https://github.com/OpenNMS/opennms/blob/develop/opennms-config-model/src/main/java/org/opennms/netmgt/config/syslogd/Configuration.java .If we need to change then we are ready to update it.

Reply:
In the case referenced we perform validation on *all* of the attributes, whereas here we only verify that the `threads` field is not null. The behavior is not consistent.